app/routes/rental_routes.py

```diff
@@ -0,0 +1,98 @@
+from flask import Blueprint, jsonify, make_response, request, abort
+from flask.helpers import make_response
+from app.models.rental import Rental
+from app.models.customer import Customer
+from app.models.video import Video
+from app import db
+from datetime import datetime, timedelta
+from app.common_functions.check_for_id import valid_int, get_id
+from app.common_functions.check_request_body import chek_request_body
+
+rental_bp = Blueprint("rental", __name__, url_prefix="/rentals")
+
+OVERDUE_VALID_SORTS = ["title", "name", "checkout_date", "due_date"]
+
+#POST/RENTALS/CHECK-OUT
+
+@rental_bp.route("/check-out", methods=["POST"])
+def create_rental():
+    request_body = request.get_json()
+    
+    request_body_parameters = ["customer_id", "video_id"]
+    chek_request_body(request_body_parameters) 
+
+    get_id(request_body["customer_id"],Customer, str_repr="Customer")
+    get_id(request_body["video_id"], Video, str_repr="Video")
+
+    new_rental = Rental(
+        customer_id = request_body["customer_id"],
+        video_id = request_body["video_id"], 
+        due_date = datetime.utcnow() + timedelta(7)
+    )
+
+    db.session.add(new_rental)
+    db.session.commit()
+
+    
+    avilable_inventory = new_rental.get_available_inventory()
+   
+    if   avilable_inventory  < 0:
+        abort(make_response({"message": "Could not perform checkout."}, 400))
+
+    return make_response(new_rental.to_dict(),200)
+
+
+@rental_bp.route("/check-in", methods=["POST"])
+def create_check_in():
+    request_body = request.get_json()
+
+    request_body_parameters = ["customer_id", "video_id"]
+    chek_request_body(request_body_parameters)
+    
+    video_id = request_body["video_id"]
+    customer_id = request_body["customer_id"]
+
+    
+    check_customer = get_id(customer_id, Customer, "customer")
+    check_video = get_id(video_id, Video, "video")
+    
+    rental_to_check_in = Rental.query.filter(
+        Rental.video_id == video_id,
+        Rental.customer_id == customer_id,
+        Rental.checked_in.is_(None)
+        ).first()
+    
+    if not rental_to_check_in:
+        return make_response({"message": f"No outstanding rentals for customer {customer_id} and video {video_id}"}, 400)
+    
+    rental_to_check_in.checked_in = datetime.utcnow()
```

Comment:
You might consider moving this logic to a class method on the rental that either returns the `rental` to be checked-in or `None` if it is not found. In this way, we keep the route function focused on processing the input and building the output, and put the more complex logic with the model.